src/main/kotlin/net/casual/arcade/border/MovingCenterBorderState.kt

```diff
@@ -33,13 +39,13 @@ class MovingCenterBorderState(
                 // We do not want to update DelegateBorderChangeListener
                 // This updates borders in other dimensions
                 if (listener !is BorderChangeListener.DelegateBorderChangeListener) {
-                    listener.onBorderCenterSet(this.border, this.center.x, this.center.y)
+                    listener.onBorderCenterSet(this.border, this.getCenterX(), this.getCenterZ())
 
                 }
             }
         }
 
-        return if (this.ticks >= this.tickDuration) StillCenterBorderState(this.center) else this
+        return if (this.ticks >= this.tickDuration) StillCenterBorderState(this.centerX, this.centerZ) else this
```

Comment:
When you reach the end this should be `this.targetCenterX` and `this.targetCenterZ` otherwise you're setting it back to the start

Reply:
I think I need to actually call the lerp on it to start from where it is currently.

Reply:
wait, you actually may be right. as this is when it is over...